cache/clock_cache.h

```diff
@@ -45,25 +44,27 @@ namespace clock_cache {
 //    Importantly, a handle can be evicted if and only if it's not
 //    referenced. In particular, when an handle becomes referenced, it's
 //    temporarily taken out of clock until all references to it are released.
-// (M) Marked for deletion (or invisible): An handle is marked for deletion
-//    when an operation attempts to delete it, but the handle is externally
-//    referenced, so it can't be immediately deleted. When this mark is placed,
-//    lookups will no longer be able to find it. Consequently, no more external
-//    references will be taken to the handle. When a handle is marked for
-//    deletion, we also say it's invisible.
+// (M) Marked for deletion (or invisible): An handle becomes marked for deletion
+//    when an operation tries to delete it. When the handle is externally
+//    referenced, it can't be immediately deleted and the attempt fails.
+//    In these cases, the mark will be used later by the eviction algorithm.
+//    When this mark is placed, lookups are no longer be able to find the
+//    handle. Consequently, no more external references can be taken to the
+//    handle. For this reason, When a handle is marked for deletion, we also
+//    say it's invisible.
 // These properties induce 4 different states, with transitions defined as
 // follows:
 // - Not M --> M: When a handle is deleted or replaced by a new version, but
 //    not immediately evicted.
 // - M --> not M: This cannot happen. Once a handle is marked for deletion,
-//    there is no can't go back.
+//    there is no way back.
```

Comment:
If possible, it'd be good to document the transitions in more detail, including the triggers for transition. For example, what would cause a handle to be marked for deletion but not immediately evicted, transition from internal to external ref or vice versa.

Reply:
I will work on improving this part of the documentation.